aksel.nav.no/website/components/website-modules/ConsentBanner.tsx

```diff
@@ -0,0 +1,162 @@
+"use client";
+
+import Link from "next/link";
+import Script from "next/script";
+import { useEffect, useRef, useState } from "react";
+import { Cookies } from "typescript-cookie";
+import { BodyLong, Button, Modal } from "@navikt/ds-react";
+import { classifyTraffic } from "../utils/get-current-environment";
+
+const CONSENT_TRACKER_ID = "aksel-consent";
+
+type CONSENT_TRACKER_STATE =
+  | "undecided"
+  | "accepted"
+  | "rejected"
+  | "no_action";
+
+type CookieData = {
+  createdAt: string;
+  updatedAt: string;
+  version: number;
+  consents: {
+    tracking?: CONSENT_TRACKER_STATE;
+  };
+};
+
+export const getStorageAcceptedTracking = () => {
+  const rawState = (Cookies.get(CONSENT_TRACKER_ID) ??
+    sessionStorage.getItem(CONSENT_TRACKER_ID)) as string;
+
+  if (!rawState) {
+    return "undecided";
+  }
+
+  const cookieData = JSON.parse(rawState) as CookieData;
+
+  return cookieData.consents.tracking as CONSENT_TRACKER_STATE;
+};
+
+export const setStorageAcceptedTracking = (state: CONSENT_TRACKER_STATE) => {
+  const cookieData: CookieData = {
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    version: 1,
+    consents: {},
+  };
+
+  cookieData.consents.tracking = state;
+
+  const cookieJson = JSON.stringify(cookieData);
+
+  if (state === "no_action") {
+    sessionStorage.setItem(CONSENT_TRACKER_ID, cookieJson);
+    return;
+  }
+
+  Cookies.set(CONSENT_TRACKER_ID, cookieJson, {
+    expires: 365,
+    domain: "aksel.ansatt.dev.nav.no",
+  });
+};
+
+/*
+ * close handler will always run after both clicking a button or the 'X'
+ * therefore we have to check if the user closed via one of the buttons
+ * (a cookie or sessionStorage was set) or if it was simply closed via the 'X' (no action taken)
+ * this is where we can then act on "no action taken"*/
+// NOTE: maybe we should have a specific handler for "no action",
```

Comment:
Is this a good idea? 🤔 (adding a specific handler function for the "x" ?)

Reply:
no longer an issue, since we removed modal, but might still be an idea for the modal user-facing API?